src/KawaBot.Server/OAuthController.fs

```diff
@@ -0,0 +1,97 @@
+﻿namespace KawaBot.Server
+
+open System.IO
+open Http
+open System.Net.Http
+open System.Net
+open Newtonsoft.Json
+open Environment
+open System.Collections.Generic
+open Newtonsoft.Json.Linq
+open System.Text
+open System.Threading.Tasks
+open System
+
+[<JsonObject>]
+type SlackError =
+    {
+        [<field: JsonProperty("ok") >]
+        Ok: bool;
+        [<field: JsonProperty("error") >]
+        Error: string;
+    }
+
+[<JsonObject>]
+type OAuthBot = {
+    [<field: JsonProperty("bot_user_id") >]
+    BotUserId: string;
+    [<field: JsonProperty("bot_access_token") >]
+    BotAccessToken: string;
+}
+
+[<JsonObject>]
+type OAuthSuccessResponse = 
+    {
+        [<field: JsonProperty("ok") >]
+        Ok: bool;
+        [<field: JsonProperty("access_token") >]
+        AccessToken: string;
+        [<field: JsonProperty("scope") >]
+        Scope: string;
+        [<field: JsonProperty("user_id") >]
+        UserId: string;
+        [<field: JsonProperty("team_name") >]
+        TeamName: string;
+        [<field: JsonProperty("team_id") >]
+        TeamId: string;
+        [<field: JsonProperty("bot") >]
+        Bot: OAuthBot;
+    }
+
+module OAuthHelper =
+    let GetOAuthURL(code: string): string =
+            let urlArgs = [
+                "https://slack.com/api/oauth.access";
+                sprintf "?client_id=%s" Environment.ClientID;
+                sprintf  "&client_secret=%s" Environment.ClientSecret;
+                sprintf "&code=%s" code;
+            ] 
+            urlArgs |> String.concat ""
+
+type IHTTPRequester = 
+    abstract member HTTPGet: url: string -> HttpResponseMessage
```

Comment:
Lets use the Pascal case for abbreviations longer than 3 chars: `HTTP` -> `Http`.